File: Kth Smallest Element in a BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int kthsmall(TreeNode* root, int &count , int k)
    {
        if (root==NULL)
            return -1;
        int left= kthsmall(root->left,count,k);
        if(left!=-1)
            return left;
        count++;
        if(count==k)
        {
            return root->val;
        }
        return kthsmall(root->right,count,k);
    }
    int kthSmallest(TreeNode* root, int k) {
        int count=0;
        return kthsmall(root,count,k);
        
    }
};

File: Golang Solutions/Kth Smallest Element in a BST.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func kthsmall(root *TreeNode,count *int, k int) int {
    if root==nil{
        return -1       
    }   
    var left int
    left= kthsmall(root.Left,count,k)
    if left!=-1{
        return left
    }
    *count++        
    if *count==k{
        return root.Val
    }
    return kthsmall(root.Right,count,k);
}
func kthSmallest(root *TreeNode ,  k int)int {
    count:=0
    return kthsmall(root,&count,k)
}